src/Compiler/TypedTree/TypedTreeOps.fs

```diff
@@ -10360,3 +10360,24 @@ let isFSharpExceptionTy g ty =
     | ValueSome tcref -> tcref.IsFSharpException
     | _ -> false
 
+let (|EmptyModuleOrNamespace|_|) (moduleOrNamespaceContents: ModuleOrNamespaceContents) =
+    match moduleOrNamespaceContents with
+    | TMDefs(defs = defs) ->
+        let singleModuleOrNamespace =
+            defs
+            |> List.choose (function
+                | ModuleOrNamespaceContents.TMDefRec _ as defRec
+                | ModuleOrNamespaceContents.TMDefs(defs = [ ModuleOrNamespaceContents.TMDefRec _ as defRec ]) -> Some defRec
+                | _ -> None)
+            |> List.tryHead
```

Comment:
What if there are multiple items in this list?

Reply:
That is a very unlikely scenario I believe.
You can only have this with two namespaces.
Fair enough, I've added a test.